src/Symfony/Component/Clock/DatePoint.php

```diff
@@ -66,6 +66,33 @@ public static function createFromMutable(\DateTime $object): static
         return parent::createFromMutable($object);
     }
 
+    public static function createFromTimestamp(int|float $timestamp): static
+    {
+        if (\PHP_VERSION_ID >= 80400) {
+            return parent::createFromTimestamp($timestamp);
+        }
+
+        if (\is_int($timestamp)) {
+            return static::createFromFormat('U', (string) $timestamp);
+        }
+
+        if (!is_finite($timestamp) || \PHP_INT_MAX + 1.0 <= $timestamp || \PHP_INT_MIN > $timestamp) {
+            throw new \DateRangeError(sprintf('DateTimeImmutable::createFromTimestamp(): Argument #1 ($timestamp) must be a finite number between %s and %s.999999, %s given', \PHP_INT_MIN, \PHP_INT_MAX, $timestamp));
+        }
+
+        if (0 <= $timestamp) {
+            return static::createFromFormat('U.u', sprintf('%.6F', $timestamp));
+        }
+
+        $microsecond = 1_000_000 - (int) (((int) $timestamp - $timestamp) * 1_000_000);
+        if (1_000_000 === $microsecond) {
+            $microsecond = 0;
+        }
+
+        return static::createFromFormat('U', (string) (int) floor($timestamp))
+            ->setMicrosecond($microsecond);
```

Comment:
we can also optimize for negative timestamps:

```suggestion
        if (\is_int($timestamp) || !$ms = (int) $timestamp - $timestamp) {
            return static::createFromFormat('U', (string) $timestamp);
        }

        if (!is_finite($timestamp) || \PHP_INT_MAX + 1.0 <= $timestamp || \PHP_INT_MIN > $timestamp) {
            throw new \DateRangeError(sprintf('DateTimeImmutable::createFromTimestamp(): Argument #1 ($timestamp) must be a finite number between %s and %s.999999, %s given', \PHP_INT_MIN, \PHP_INT_MAX, $timestamp));
        }

        if ($timestamp < 0) {
            $timestamp = (int) $timestamp - 2.0 + $ms;
        }

        return static::createFromFormat('U.u', sprintf('%.6F', $timestamp));
```